firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -56,14 +58,18 @@
   @GuardedBy("updateIfNewReleaseTaskLock")
   private UpdateTaskImpl cachedUpdateIfNewReleaseTask;
 
-  private final Object cachedNewReleaseLock = new Object();
-
-  @GuardedBy("cachedNewReleaseLock")
-  private AppDistributionReleaseInternal cachedNewRelease;
+  AtomicReference<AppDistributionReleaseInternal> cachedNewRelease = new AtomicReference();
```

Comment:
Since this had a separate lock from the cached task, I thought we could reduce the number of blocks by adding an atomic reference